### ISSUE TITLE
TypeError: unsupported operand type(s) for -: 'datetime.datetime' and 'NoneType'

### ISSUE THREAD

**Issue by neequole** [celery/celery#3326]
Hi,

To make task results not to expire, I tried setting CELERY_TASK_RESULT_EXPIRES = None for Celery with Django as broker. Though the tasks were not deleted, the backend_cleanup task raised an unsupported operand type(s) for -: 'datetime.datetime' and 'NoneType' error, is this expected? Because when I setup it as 0,
![selection_025](https://cloud.githubusercontent.com/assets/2977662/16973471/7e76df16-4e66-11e6-9d3b-4a1ff78b1c87.png)
 task results are still deleted by the backend_cleanup tasks. Thanks!

**Comment by ask** [comment]
Did you restart `beat`? It wouldn't be able to notice the setting changed I think.

Also the django-celery database scheduler only imports the setting schedule when something is added, not when they are changed/removed, so if you're using the database scheduler you should go into the admin and remove that periodic task.

**Comment by neequole** [comment]
Hi Ask,

Thanks for your quick response. Yes I did restart beat and changes were reflected for CELERY_TASK_RESULT_EXPIRES setting but I got the error as mentioned above for None value. I guess I'll just go with disabling the periodic task in the admin :)

Btw, just to share, upon checking the source code behavior for python (djcelery/managers.py in get_all_expired):
1) For expires 0 value - `self.filter(date_done__lt=now() - maybe_timedelta( **0** ) ))` will consider all task results before current timestamp as expired.
2) For expires None value - `self.filter(date_done__lt=now() - maybe_timedelta( **None** ) ))` will return the TypeError because of attempt to subtract type datetime with None value.
